### DISCUSSION TITLE
Environment Variables + GCP

### DISCUSSION THREAD

**Discussion by tsaldanha** [prisma/prisma#25815] [Q&A] [answered]
### Question

I am migrating from vercel to GCP, and my nextjs build runs, but I must create my database and run the migrations. 

    "gcp-build": "prisma generate && prisma migrate deploy && next build"
    
    My enviroment variables are ok for the next build, but are the prisma command can't get then giving me an error on : 
    

```
Error code: P1012
error: Environment variable not found: POSTGRES_URL_NON_POOLING.
 --> prisma/schema.prisma:9
8 | url = env("POSTGRES_PRISMA_URL")
9 | directUrl = env("POSTGRES_URL_NON_POOLING")
```

Both variables are included on Cloud Run console. 
Next build see all the variables, just prisma generate and migrate can't see it 

What should I do? 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @tsaldanha 👋 

This may occur because the environment variables specified in the Cloud Run console are not accessible during the build step. In Cloud Run, the build process runs in a different environment than the one where the runtime environment variables are available.
Can you update your gcp-build script to load the `.env` file: Use a tool like [dotenv-cli](https://www.npmjs.com/package/dotenv-cli) to load environment variables from the .env file.

```
"gcp-build": "dotenv -e .env -- prisma generate && dotenv -e .env -- prisma migrate deploy && next build"
```

**Reply by tsaldanha** [reply]
Thanks a lot @ludralph !

Just one more thing, I need to create this .env file in my repo, exposing my vars, or this .env file is created before the build on cloud run?

Best Regards,

**Reply by ludralph** [reply, marked as answer]
Hi @tsaldanha 

Storing `.env` in the repository is not secure, as it exposes sensitive credentials and configuration information. You can use runtime secrets or inject them securely during the build process. You can use Google Secret Manager to store sensitive data.
